File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/c_solutions/222-count-complete-tree-nodes.c
int countNodes(struct TreeNode* root) {
    if (root == NULL) return 0;

    int left_height = 0, right_height = 0;
    struct TreeNode* left = root;
    struct TreeNode* right = root;

    while (left != NULL) {
        left_height++;
        left = left->left;
    }

    while (right != NULL) {
        right_height++;
        right = right->right;
    }

    if (left_height == right_height) {
        return (1 << left_height) - 1;
    }

    return 1 + countNodes(root->left) + countNodes(root->right);
}

File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/java_solutions/222-count-complete-tree-nodes.java
class Solution {
    public int countNodes(TreeNode root) {
        if (root == null) {
            return 0;
        }

        int leftHeight = 0;
        TreeNode leftNode = root;
        while (leftNode != null) {
            leftHeight++;
            leftNode = leftNode.left;
        }

        int rightHeight = 0;
        TreeNode rightNode = root;
        while (rightNode != null) {
            rightHeight++;
            rightNode = rightNode.right;
        }

        if (leftHeight == rightHeight) {
            return (1 << leftHeight) - 1;
        } else {
            return 1 + countNodes(root.left) + countNodes(root.right);
        }
    }
}